Plot removal of low quality cells in QC report (#138)
AlexsLemonade/scpcaTools: inst/rmd/qc_report.rmd

## Removing low quality cells

The below plot highlights cells that were removed prior to normalization and dimensionality reduction.
Cells that should be removed are those that are identified to be low quality cells, such as cells with high probability of being compromised, calculated by `miQC`.
The method of filtering is indicated above the plot as either `miQC` or `Minimum gene cutoff`. 
If `miQC`, cells below the specified probability compromised cutoff and above the minimum number of unique genes identified are kept for downstream analyses. 
If only a `Minimum gene cutoff` is used, then `miQC` is not used and only those cells that pass the minimum number of unique genes identified threshold are retained. 
The dotted vertical line indicates the minimum gene cutoff used for filtering. 

```{r}
if(has_filtered & has_processed){
  
  # grab cutoffs and filtering method from processed sce
  min_gene_cutoff <- processed_meta$min_gene_cutoff
  filter_method <- processed_meta$ccdl_filter_method
  
  # add column to coldata labeling cells to keep/remove based on filtering method

>   filtered_coldata_df <- colData(filtered_sce) |>
>     as.data.frame()
>     
>   filtered_coldata_df <- filtered_coldata_df |>
>     dplyr::mutate(ccdl_filter = ifelse(rownames(filtered_coldata_df) %in%  colnames(processed_sce), 
>                                        "Keep", 
>                                        "Remove")) 

Reviewer: `tidy`ing up a bit.

```suggestion
  filtered_coldata_df <- colData(filtered_sce) |>
    as.data.frame() |>
    tibble::rownames_to_column("barcode") |>
    dplyr::mutate(ccdl_filter = ifelse(barcode %in%  colnames(processed_sce), 
                                       "Keep", 
                                       "Remove")) 
```

  
  ggplot(filtered_coldata_df, aes(x = detected, y = subsets_mito_percent, color = ccdl_filter)) + 
    geom_point(alpha = 0.5, size = 1) +

>     geom_vline(xintercept = dmin_gene_cutoff, linetype = "dashed") +

Reviewer: I think a typo?
```suggestion
    geom_vline(xintercept = min_gene_cutoff, linetype = "dashed") +
```

## Removing low quality cells

The below plot highlights cells that were removed prior to normalization and dimensionality reduction.
Cells that should be removed are those that are identified to be low quality cells, such as cells with high probability of being compromised, calculated by `miQC`.
The method of filtering is indicated above the plot as either `miQC` or `Minimum gene cutoff`. 
If `miQC`, cells below the specified probability compromised cutoff and above the minimum number of unique genes identified are kept for downstream analyses. 
If only a `Minimum gene cutoff` is used, then `miQC` is not used and only those cells that pass the minimum number of unique genes identified threshold are retained. 
The dotted vertical line indicates the minimum gene cutoff used for filtering. 

```{r}
if(has_filtered & has_processed){
  
  # grab cutoffs and filtering method from processed sce
  min_gene_cutoff <- processed_meta$min_gene_cutoff
  prob_compromised_cutoff <- processed_meta$prob_compromised_cutoff
  filter_method <- processed_meta$ccdl_filter_method
  
  # add column to coldata labeling cells to keep/remove based on filtering method

>   if(filter_method == "miQC"){
>     
>     filtered_coldata_df <- filtered_coldata_df %>%
>       dplyr::mutate(ccdl_filter = ifelse(prob_compromised <= prob_compromised_cutoff & 
>                                            detected >= min_gene_cutoff, 
>                                          "Keep", 
>                                          "Remove"))
>     
>   } else if(filter_method == "Minimum_gene_cutoff") {
>     
>     filtered_coldata_df <- filtered_coldata_df %>%
>       dplyr::mutate(ccdl_filter = ifelse(detected >= min_gene_cutoff, 
>                                          "Keep", 
>                                          "Remove"))
>     
>   }

Reviewer: I think we want to simplify this to use the processed data. This also means we don't need different code for miQC & the simpler cutoff. 

We can probably move making `filtered_coldata_df` to here as well, since we won't need the `prob_compromised` at all
```suggestion
  filtered_coldata_df <- colData(filtered_sce) |>
    as.data.frame()
    
  filtered_coldata_df <- filtered_coldata_df |>
    dplyr::mutate(ccdl_filter = ifelse(rownames(filtered_coldata_df) %in%  colnames(processed_sce), 
                                       "Keep", 
                                       "Remove")) 
```

  
  ggplot(filtered_coldata_df, aes(x = detected, y = subsets_mito_percent, color = ccdl_filter)) + 
      geom_point(alpha = 0.5, size = 1) +
      geom_vline(xintercept = min_gene_cutoff, linetype = "dashed") +
      labs(x = "Number of genes detected",
           y = "Mitochondrial percentage",

>            color = "Filter") +
>     ggtitle(stringr::str_replace(filter_method, "_", " ")) +

Reviewer: Making the title part of labs?
```suggestion
           color = "Filter"
           title = stringr::str_replace(filter_method, "_", " ")) +
```

    theme(plot.title = element_text(hjust = 0.5),
          legend.position = c(1, 1),
          legend.justification = c(1,1),
          legend.background = element_rect(color = "grey20", size= 0.25),
          legend.title = element_text(hjust = 0.5))
} else {
  glue::glue("
    <div class=\"alert alert-warning\">
    

>     No filtering was performed on this library.

Reviewer: Is this accurate? Do we want to distinguish filtering from processing?
